src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/HostedMcpServerToolCallContent.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a tool call request to a hosted MCP server.
+/// </summary>
+[Experimental("MEAI001")]
+public class HostedMcpServerToolCallContent : AIContent
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HostedMcpServerToolCallContent"/> class.
+    /// </summary>
+    /// <param name="callId">The tool call ID.</param>
+    /// <param name="name">The tool name.</param>
+    /// <param name="serverName">The MCP server name.</param>
+    /// <param name="arguments">The arguments used for the tool call.</param>
+    public HostedMcpServerToolCallContent(string callId, string name, string serverName, IReadOnlyDictionary<string, object?>? arguments = null)
```

Comment:
FunctionCallContent uses IDictionary rather than IReadOnlyDictionary. Should this match?

Reply:
I don't think this is meant to be mutated, this just represents what the AI service used as arguments when invoking the tool from the MCP server, that's already been done.

Reply:
> I also wonder why the FunctionCallContent can be mutated

With FCC, it's the client doing the invocation. Middleware might examine the arguments and choose to tweak them before they're passed back, such as to remove PII, to fill in any missing information, to augment with the dictionary with additional state it knows the function may optionally want to examine, etc